ember-client/src/main/scala/org/http4s/ember/client/internal/ClientHelpers.scala

```diff
@@ -129,20 +130,27 @@ private[client] object ClientHelpers {
       .putHeaders(userAgentHeader.toSeq: _*)
   }
 
-  private[internal] def postProcessResponse[F[_]: Concurrent](
+  private[internal] def postProcessResponse[F[_]](
       req: Request[F],
       resp: Response[F],
-      canBeReused: Ref[F, Reusable]): Response[F] = {
+      drain: F[Option[Array[Byte]]],
+      nextBytes: Ref[F, Array[Byte]],
+      canBeReused: Ref[F, Reusable])(implicit F: Concurrent[F]): Response[F] = {
+    // TODO If Response Body has a take(1).compile.drain - would leave rest of bytes in root stream for next caller
     val out = resp.copy(
       body = resp.body.onFinalizeCaseWeak {
         case ExitCase.Completed =>
-          val requestClose = req.headers.get(Connection).exists(_.hasClose)
-          val responseClose = resp.headers.get(Connection).exists(_.hasClose)
+          drain.flatMap {
```

Comment:
i think we might be able to move this logic somewhere else so we can actually drain even if the body wasn't read at all.

Reply:
Stick it in the vault and move it up to where we have a `Resource[F, Response[F]]`?  Is it too late to flip the `canBeReused` at that point?

Reply:
Yep, that's what I had in mind and it worked out perfectly